### ISSUE TITLE
ListBucketMetricsConfigurationsAsync returns a empty list even though there are existing Bucket Metrics Configurations

### ISSUE THREAD

**Issue by sannidhiCimpress** [aws/aws-sdk-net#2532]
### Describe the bug

what works : 

- AWS CLI command, example:
  `aws s3api list-bucket-metrics-configurations --bucket "my_bucket" --profile my-profile`
- returns a list of metric configurations

what does not work:

- same cli command when executed using SDK with same aws credentials profile, sample code:
`        var request = new ListBucketMetricsConfigurationsRequest()
        {
            BucketName = "my_bucket"
        };
        var response  = await _s3Client.ListBucketMetricsConfigurationsAsync(request);`
- returns an empty list of list metric configurations
- whereas getting a specific metrics configuration using `GetBucketMetricsConfigurationAsync` on the same bucket returns a metric configuration of matching metricsId 

### Expected Behavior

ListBucketMetricsConfigurationsAsync returns list of MetricsConfigurations

### Current Behavior

ListBucketMetricsConfigurationsAsync returns a empty list of MetricsConfigurations, even if MetricsConfigurations exists

### Reproduction Steps

- creating a bucket metric config
- listing metrics configs using CLI command (works)
- listing metrics configs using ListBucketMetricsConfigurationsAsync returns a empty list, instead of list with created metric config

### AWS .NET SDK and/or Package version used

3.7.102.2

### Targeted .NET Platform

.NET Core 7

### Operating System and version

OSX Big Sur

**Comment by ashishdhingra** [comment]
Reproducible.
- Using AWS CLI (replace `<<bucket-name>>` with actual bucket name), 
  - create bucket metrics configuration using command `aws s3api put-bucket-metrics-configuration --bucket <<bucket-name>> --id metrics-config-id --metrics-configuration '{"Id":"metrics-config-id","Filter":{"Prefix":"prefix1"}}'`
  - Execute the AWS CLI command `aws s3api list-bucket-metrics-configurations --bucket <<bucket-name>>`. This gives the below output:
```JSON
{
    "IsTruncated": false,
    "MetricsConfigurationList": [
        {
            "Id": "metrics-config-id",
            "Filter": {
                "Prefix": "prefix1"
            }
        }
    ]
}
```
- Execute the below C# code:
```C#
var _client = new AmazonS3Client(); // Region is set in default profile
var bucketName = "<<bucket-name>>";
var listBucketMetricsConfigurationsRequest = new ListBucketMetricsConfigurationsRequest() { BucketName = bucketName };
var listBucketMetricsConfigurationsResponse = _client.ListBucketMetricsConfigurationsAsync(listBucketMetricsConfigurationsRequest).Result;
Console.WriteLine(listBucketMetricsConfigurationsResponse.MetricsConfigurationList.Count);
```
this returns `0` count.

**Comment by ashishdhingra** [comment]
**ROOT CAUSE:** The [ListBucketMetricsConfigurationsResponseUnmarshaller.UnmarshallResult(XmlUnmarshallerContext context, ListBucketMetricsConfigurationsResponse response)](https://github.com/aws/aws-sdk-net/blob/676ede89d167657c609707840edba4b57d2292a0/sdk/src/Services/S3/Custom/Model/Internal/MarshallTransformations/ListBucketMetricsConfigurationsResponseUnmarshaller.cs#L68) is testing context against `InventoryConfiguration`. It should instead be `MetricsConfiguration`. Tested it locally and it works.

**Comment by normj** [comment]
Fix has been released as part of version 3.7.103.7 of AWSSDK.S3
